### PR TITLE
AzureRT S52 PR #1

### PR DESCRIPTION
**Pull request by huangpf**
- Fix & Test Set-AzureDeployment with -ExtensionConfiguration
- Fix & Test NextLink

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@huangpf__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (phuang). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by huangpf** [comment]
retest this please

**Comment by huangpf** [comment]
retest this please

**Comment by amarzavery** [comment]
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/79/

File: src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineTests.cs [hunk 1]
```diff
@@ -116,5 +116,12 @@ public void TestGetVMSizeFromAllLocations()
         {
             ComputeTestController.NewInstance.RunPsTest("Test-GetVMSizeFromAllLocations");
         }
+
+        [Fact(Skip = "Fix NextLink")]
```
Comment (amarzavery) [inline src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineTests.cs]: Why we want to skip it? Your PR comment claims you are fixing the "nextlink"?

Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineTests.cs]: The client library has some problems. Will be updating the client & test soon.

**Comment by huangpf** [comment]
@amarzavery Everything is included now. Please help review. Thanks.

**Comment by huangpf** [comment]
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/81/

**Comment by huangpf** [comment]
retest this please

**Comment by huangpf** [comment]
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/105/console

**Comment by stankovski** [comment]
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-msi/413/

**Comment by stankovski** [comment]
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/116/

**Pull request merged**